instrumentation/google.golang.org/grpc/otelgrpc/filters/filters.go

```diff
@@ -0,0 +1,158 @@
+// returns true if any Filter in the list returns true.
+func Any(fs ...otelgrpc.Filter) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		for _, f := range fs {
+			if f(i) {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+// All takes a list of Filters and returns a Filter that
+// returns true only if all Filters in the list return true.
+func All(fs ...otelgrpc.Filter) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		for _, f := range fs {
+			if !f(i) {
+				return false
+			}
+		}
+		return true
+	}
+}
+
+// None takes a list of Filters and returns a Filter that returns
+// true only if none of the Filters in the list return true.
+func None(fs ...otelgrpc.Filter) otelgrpc.Filter {
+	return Not(Any(fs...))
+}
+
+// Not provides a convenience mechanism for inverting a Filter.
+func Not(f otelgrpc.Filter) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		return !f(i)
+	}
+}
+
+// MethodName returns a Filter that returns true if the request's
+// method name matches the provided string n.
+func MethodName(n string) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		p := splitFullMethod(i)
+		return p.method == n
+	}
+}
+
+// MethodPrefix returns a Filter that returns true if the request's
+// method starts with the provided string pre.
+func MethodPrefix(pre string) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		p := splitFullMethod(i)
+		return strings.HasPrefix(p.method, pre)
+	}
+}
+
+// FullMethodName returns a Filter that returns true if the request's
+// full RPC method string, i.e. /package.service/method, starts with
+// the provided string n.
+func FullMethodName(n string) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		var fm string
+		switch i.Typ {
+		case otelgrpc.UnaryClient, otelgrpc.StreamClient:
+			fm = i.Method
+		case otelgrpc.UnaryServer:
+			fm = i.UnaryServerInfo.FullMethod
+		case otelgrpc.StreamServer:
+			fm = i.StreamServerInfo.FullMethod
+		default:
+			fm = i.Method
+		}
+		return fm == n
+	}
+}
+
+// ServiceName returns a Filter that returns true if the request's
+// service name, i.e. package.service, matches s.
+func ServiceName(s string) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		p := splitFullMethod(i)
+		return p.service == s
+	}
+}
+
+// ServicePrefix returns a Filter that returns true if the request's
+// service name, i.e. package.service, starts with the provided string pre.
+func ServicePrefix(pre string) otelgrpc.Filter {
+	return func(i *otelgrpc.InterceptorInfo) bool {
+		p := splitFullMethod(i)
+		return strings.HasPrefix(p.service, pre)
+	}
+}
+
+// NotHealthCheck returns a Filter that returns true if the request's
+// is not health check defined by gRPC Health Checking Protocol.
+// https://github.com/grpc/grpc/blob/master/doc/health-checking.md
+func NotHealthCheck() otelgrpc.Filter {
```

Comment:
Thank you for the comment! For readability perspective, I agree with your point and the shortcut I wanted is too small (i.e. `NotHealthCheck()` instead of `Not(HealthCheck())`), so I change this part as you suggest.